hazelcast/src/main/resources/hazelcast-config-3.5.xsd

```diff
@@ -655,6 +656,20 @@
             </xs:simpleType>
         </xs:attribute>
     </xs:complexType>
+    <xs:complexType name="ringbuffer">
+        <xs:all>
+            <xs:element name="capacity" type="xs:unsignedInt" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="retention-seconds" type="xs:unsignedInt" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="backup-count" type="backup-count" minOccurs="0" maxOccurs="1" default="1"/>
+            <xs:element name="async-backup-count" type="backup-count" minOccurs="0" maxOccurs="1" default="0"/>
+            <xs:element name="in-memory-format" type="in-memory-format" minOccurs="0" maxOccurs="1" default="BINARY"/>
```

Comment:
I'm still not understanding because the Config objects throw NPE or IllegalArgumentException. 

For example from ExecutorConfig:

```
  /**
     * @param poolSize the poolSize to set
     */
    public ExecutorConfig setPoolSize(final int poolSize) {
        if (poolSize <= 0) {
            throw new IllegalArgumentException("poolSize must be positive");
        }
        this.poolSize = poolSize;
        return this;
    }
```

It should be a concern of the XML parser to trap all IllegalArgument/NullPointerExceptions and convert them to a InvalidConfiguraitonException.

So I'm not seeing what that has to do with my config objects.